lib/FieldType/Mapper/TextLineFormMapper.php

```diff
@@ -59,20 +67,33 @@ public function mapFieldDefinitionForm(FormInterface $fieldDefinitionForm, Field
             );
     }
 
+    /**
+     * @param FormInterface $fieldForm
+     * @param FieldData $data
+     */
     public function mapFieldValueForm(FormInterface $fieldForm, FieldData $data)
     {
         $fieldDefinition = $data->fieldDefinition;
+        $validatorConfiguration = $fieldDefinition->getValidatorConfiguration();
         $formConfig = $fieldForm->getConfig();
-        $names = $fieldDefinition->getNames();
-        $label = $fieldDefinition->getName($formConfig->getOption('languageCode')) ?: reset($names);
 
         $fieldForm
             ->add(
                 $formConfig->getFormFactory()->createBuilder()
                     ->create(
                         'value',
                         TextType::class,
-                        ['required' => $fieldDefinition->isRequired, 'label' => $label]
+                        [
+                            'required' => $fieldDefinition->isRequired,
+                            'label' => $fieldDefinition->getName($formConfig->getOption('languageCode')),
+                            'attr' => [
+                                'pattern' => sprintf(
```

Comment:
also, if the field definition has no length constraint, this will generate `pattern=".{,}"` which is bit weird (and I'm unsure of the consequences in browsers)